### PR TITLE
Add ROS2 support

### PR DESCRIPTION
**Pull request by LaurieCheers-unity**

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <package format="2">
   <name>ros_tcp_endpoint</name>
-  <version>0.4.0</version>
+  <version>0.5.0</version>
```
Comment (peifeng-unity) [inline package.xml:4]: do we only upgrade the package version when merging from `dev` to `main`?

Comment (LaurieCheers-unity) [reply package.xml:4]: We're about to release it this week, so it doesn't seem worth being nitpicky about it :-)

Comment (peifeng-unity) [reply package.xml:4]: Haha, sure; it is not important.

**Review by peifeng-unity — APPROVED** [review]
LGTM. Approve assuming the test can pass :)

**Pull request merged**